tracing-appender/src/rolling.rs

```diff
@@ -96,10 +99,15 @@ pub struct RollingFileAppender {
 ///
 /// [writer]: std::io::Write
 /// [`MakeWriter`]: tracing_subscriber::fmt::writer::MakeWriter
-#[derive(Debug)]
-pub struct RollingWriter<'a>(RwLockReadGuard<'a, File>);
+pub struct RollingWriter<'a>(MutexGuard<'a, Box<dyn Write + Send>>);
+// I've changed the above from `ReadGuard` to `MutexGuard`. Please double check if it's ok
```

Comment:
This change might have some performance consequences. The previous design used a `RwLock` to take advantage of the fact that `&File` implements the `Write` trait in order to allow multiple threads to write log entries to the file without having to contend for an exclusive write lock. Instead, a write lock is only needed when the log file is being _changed_ to a new file, which happens much less frequently than writing a log entry. Changing this from an `RwLock` to a `Mutex` means that any time multiple threads attempt to write a log entry concurrently, they are forced to synchronize with each other, which feels pretty unfortunate.